README.md

```diff
@@ -25,113 +24,123 @@ Puppet Strings generates HTML documentation for Puppet extensions written in Pup
 
-For Puppet 3.x:
+   To specify a different file, use the `--out` option and specify the path and filename:
 
-```
-$ puppet resource package puppet-strings provider=gem
-```
+   ```
+   puppet strings generate --format markdown --out docs/INFO.md
+   ```
 
-For Puppet Enterprise 3.8 (Linux):
+### Generate documentation in JSON
 
-```
-GEM_HOME=/opt/puppet/lib/ruby/gems/1.9.1 puppet resource package puppet-strings provider=gem
-```
+Strings can generate a `.json` file or print JSON to stdout. This can be useful for handling or displaying the data with your own custom applications.
 
-For Puppet Enterprise 3.8 (Windows)
+By default, Strings prints JSON output to stdout.
 
-```
-$env:GEM_HOME = "C:\Program Files\Puppet Labs\Puppet Enterprise\sys\ruby\lib\ruby\gems\2.0.0"
-puppet resource package puppet-strings provider=gem
-```
+1. Change directory into the module: `cd /modules/<MODULE_NAME>`.
+2. Run the command: `puppet strings generate --format json`.
 
-### 4. Optional: Set YARD options for Strings
+   To generate JSON documentation to a file instead, use the `--out` option and specify a filename:
+   
+   ```
+   puppet strings generate --format json --out documentation.json
+   ```
 
-Puppet Strings supports YARD options (on the command line, run `yard help doc` for a list of possible options. To set YARD options, specify a `yardopts` file in the same directory in which you run `puppet strings`. 
+For details about Strings JSON output, see [Strings JSON schema](https://github.com/puppetlabs/puppet-strings/blob/master/JSON.md).
 
-Puppet Strings supports the Markdown format and automatically sets the YARD `markup` option to `markdown`.
+## Reference
 
-## Generating documentation with Puppet Strings
 
-The `puppet strings` command processes the `README` file, as well as all Puppet and Ruby source files under the `./manifests/`, `./functions/`, and `./lib/`
-directories, and then creates HTML documentation under the `./doc/` directory.
+The `puppet strings` command generates module documentation based on code and code comments. 
 
-### Generate module documentation
+By default, running `puppet strings` generates HTML documentation for a module into a `docs` directory within that module. To pass any options or arguments, use the `generate` action.
 
-To generate documentation for a Puppet module, run Strings from that module's directory. 
+Action   | Description   
+----------------|-------------------------
 
-1. Change directory into the module: `cd /modules/sample-module`.
-2. Run the command: `puppet strings`.
+`generate` | Generates documentation with the specified parameters, including format and output location.
+`server` | Serves documentation for all modules in the [module path](https://docs.puppet.com/puppet/latest/reference/dirs_modulepath.html) locally at `http://localhost:8808`.
 
-### Generate documentation for specific files
+### `puppet strings generate` command reference
```

Comment:
do we want to get into maintaining a command reference in the README? i wonder if we could generate this section :D

Reply:
We don't, no. This is pretty much just where I have it for now. I'll pull it out once I have it in the docs